packages/admin/admin/src/tabs/TabScrollButton.tsx

```diff
@@ -1,51 +1,58 @@
 import { ChevronLeft, ChevronRight } from "@comet/admin-icons";
-import {
-    ButtonBase,
-    ComponentsOverrides,
-    TabScrollButtonClassKey as MuiTabScrollButtonClassKey,
-    TabScrollButtonProps as MuiTabScrollButtonProps,
-    Theme,
-} from "@mui/material";
-import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import { ButtonBase, ButtonBaseProps, ComponentsOverrides } from "@mui/material";
+import { css, styled, Theme, useThemeProps } from "@mui/material/styles";
 import isMobile from "is-mobile";
 import * as React from "react";
 
-export type TabScrollButtonClassKey = MuiTabScrollButtonClassKey;
+export type TabScrollButtonClassKey = "root" | "vertical" | "disabled";
 
-export interface TabScrollButtonProps extends MuiTabScrollButtonProps {
-    onClick: (event: React.MouseEvent<HTMLElement>) => void;
+type OwnerState = Pick<TabScrollButtonProps, "orientation">;
+
+const Root = styled(ButtonBase, {
+    name: "CometAdminTabScrollButton",
+    slot: "root",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [styles.root, ownerState.orientation && styles.vertical];
+    },
+})<{ ownerState: OwnerState }>(
+    ({ ownerState }) => css`
+        width: 40px;
+        flex-shrink: 0;
+
+        ${ownerState.orientation === "vertical" &&
+        css`
+            width: 100%;
+            height: 40px;
+        `}
+
+        &:disabled {
+            opacity: 0.25;
+        }
+    `,
+);
+
+export interface TabScrollButtonProps extends ButtonBaseProps {
+    orientation: "horizontal" | "vertical";
+    direction: "left" | "right";
 }
 
-const styles = () => {
-    return createStyles<TabScrollButtonClassKey, TabScrollButtonProps>({
-        root: {
-            width: 40,
-            flexShrink: 0,
-        },
-        vertical: {
-            width: "100%",
-            height: 40,
-        },
-        disabled: {
-            opacity: 0.25,
-        },
+export function TabScrollButton(inProps: TabScrollButtonProps) {
+    const { orientation, direction, disabled, onClick, ...restProps } = useThemeProps({
```

Comment:
You can remove `disabled` and `onClick` here, as they are directly passed into the `Root` slot, they will be added by `...restProps` anyway.